internal/netsync/manager.go

```diff
@@ -1574,6 +1574,10 @@ func (m *SyncManager) handleInvMsg(imsg *invMsg) {
 // evicting a random value if adding the new value would cause it to
 // overflow the maximum allowed.
 func limitAdd(m map[chainhash.Hash]struct{}, hash chainhash.Hash, limit int) {
+	// Nothing to do if entry is already in the map.
+	if _, exists := m[hash]; exists {
```

Comment:
This existing behavior is intentional.  It's acting sort of like an LRU, but with random entry eviction when the limit is reached instead of the oldest.

Note that the aforementioned behavior only applies to headers, because the other usages for transactions and mixing messages do this check before calling this method.

Reply:
Looking at this again, this will be fine after the change I made in #3444 since it no longer calls this method and the other cases essentially already do this anyway.